qlib/Mime.qm

```diff
@@ -1352,4 +1413,185 @@ hash: (2 members)
             return getMsgAndHeadersIntern(NOTHING, False);
         }
     }
+
+    #! implements the MultiPartFormDataMessage class, a specialization of MultiPartMessage
+    /** @since %Mime 1.4
+    */
+    public class MultiPartFormDataMessage inherits MultiPartMessage {
+        private {
+            #! hash of part names to ensure uniqueness
+            hash parts;
+        }
+
+        #! creates the object
+        /** @param boundary the boundary to use between parts
+         */
+        constructor(string boundary = MultiPartMessage::getBoundary()) : MultiPartMessage(MPT_FORM_DATA, boundary) {
+        }
+
+        #! adds a message part to the start of the list
+        /** @param data the data for the part
+            @param name the unique name for the part
+            @param hdr an optional hash of headers for the part
+
+            @throw MULTIPART-FORMDATA-ERROR the name parameter is empty or conflicts with an existing part name
+         */
+        splicePart(data data, string name, *hash hdr) {
```

Comment:
Could you change the `data` variable name so it doesn't collide with the `data` type?

Reply:
done